dev/system-test/firestore.ts

```diff
@@ -1647,12 +1685,72 @@ describe('Query class', () => {
       });
   });
 
-  it('has limit() method', async () => {
+  it('can run get() on empty collection', async () => {
+    return randomCol.get().then(res => {
+      return expect(res.empty);
+    });
+  });
+
+  it('can run stream() on empty collection', done => {
+    let received = 0;
+    return randomCol
+      .stream()
+      .on('data', () => {
+        ++received;
+      })
+      .on('end', () => {
+        expect(received).to.equal(0);
+        done();
+      });
+  });
+
+  it('has limit() method on get()', async () => {
     await addDocs({foo: 'a'}, {foo: 'b'});
     const res = await randomCol.orderBy('foo').limit(1).get();
     expectDocs(res, {foo: 'a'});
   });
 
+  it('has limit() method on stream()', done => {
+    let received = 0;
+    Promise.all([addDocs({foo: 'a'}, {foo: 'b'})]).then(() => {
```

Comment:
Wrapping this statement in `Promise.all(...)` is misleading, and I don't think it adds value. Consider instead

```
addDocs(...).then(...
```